src/hotspot/os/linux/os_linux.cpp

```diff
@@ -289,6 +289,29 @@ julong os::Linux::free_memory() {
   return free_mem;
 }
 
+jlong os::total_swap_space() {
+  if (OSContainer::is_containerized()) {
+    return (jlong)(OSContainer::memory_and_swap_limit_in_bytes() - OSContainer::memory_limit_in_bytes());
+  } else {
+    struct sysinfo si;
+    int ret = sysinfo(&si);
+    if (ret != 0) {
+      return -1;
+    }
+    return  (jlong)(si.totalswap * si.mem_unit);
+  }
+}
+
+jlong os::free_swap_space() {
+  // TODO support free swap space in container APIs
+  struct sysinfo si;
+  int ret = sysinfo(&si);
+  if (ret != 0) {
+    return -1;
+  }
+  return (jlong)(si.freeswap * si.mem_unit);
+}
```

Comment:
In a containerized environment with some memory limit this could potentially return a large value for `free_swap_space()`, and a small(er) value for `total_swap_space()`. i.e. `total_swap_space() < free_swap_space()`. Please return `-1` if the containerized value is not supported. Better yet, push the implementation to `Linux::free_swap_space()` and `Linux::total_swap_space()` which always returns host swap values and do the (container) wrappers here in `os::free_swap_space()` and `os::total_swap_space()`.

Reply:
> In a containerized environment with some memory limit this could potentially return a large value for `free_swap_space()`, and a small(er) value for `total_swap_space()`. i.e. `total_swap_space() < free_swap_space()`. 

Yes this is what we see in our tests last night (after my change).

>Please return `-1` if the containerized value is not supported. 

okay I can do this.